lib/ansible/modules/cloud/oracle/oci_vcn.py

```diff
@@ -0,0 +1,223 @@
+        }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.oracle import oci_utils
+
+try:
+    from oci.core.virtual_network_client import VirtualNetworkClient
+    from oci.core.models import CreateVcnDetails
+    from oci.core.models import UpdateVcnDetails
+
+    HAS_OCI_PY_SDK = True
+except ImportError:
+    HAS_OCI_PY_SDK = False
+
+
+def delete_vcn(virtual_network_client, module):
+    result = oci_utils.delete_and_wait(
+        resource_type="vcn",
+        client=virtual_network_client,
+        get_fn=virtual_network_client.get_vcn,
+        kwargs_get={"vcn_id": module.params["vcn_id"]},
+        delete_fn=virtual_network_client.delete_vcn,
+        kwargs_delete={"vcn_id": module.params["vcn_id"]},
+        module=module,
+    )
+    return result
+
+
+def update_vcn(virtual_network_client, module):
+    result = oci_utils.check_and_update_resource(
+        resource_type="vcn",
+        client=virtual_network_client,
+        get_fn=virtual_network_client.get_vcn,
+        kwargs_get={"vcn_id": module.params["vcn_id"]},
+        update_fn=virtual_network_client.update_vcn,
+        primitive_params_update=["vcn_id"],
+        kwargs_non_primitive_update={UpdateVcnDetails: "update_vcn_details"},
+        module=module,
+        update_attributes=UpdateVcnDetails().attribute_map.keys(),
+    )
+    return result
+
+
+def create_vcn(virtual_network_client, module):
+    create_vcn_details = CreateVcnDetails()
+    for attribute in create_vcn_details.attribute_map.keys():
+        if attribute in module.params:
+            setattr(create_vcn_details, attribute, module.params[attribute])
+
+    result = oci_utils.create_and_wait(
+        resource_type="vcn",
+        create_fn=virtual_network_client.create_vcn,
+        kwargs_create={"create_vcn_details": create_vcn_details},
+        client=virtual_network_client,
+        get_fn=virtual_network_client.get_vcn,
+        get_param="vcn_id",
+        module=module,
+    )
+    return result
+
+
+def main():
+    module_args = oci_utils.get_taggable_arg_spec(
+        supports_create=True, supports_wait=True
+    )
+    module_args.update(
+        dict(
+            cidr_block=dict(type="str", required=False),
+            compartment_id=dict(type="str", required=False),
+            display_name=dict(type="str", required=False, aliases=["name"]),
+            dns_label=dict(type="str", required=False),
+            state=dict(
+                type="str",
+                required=False,
+                default="present",
+                choices=["absent", "present"],
+            ),
+            vcn_id=dict(type="str", required=False, aliases=["id"]),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False,
+        mutually_exclusive=[["compartment_id", "vcn_id"]],
+    )
+
+    if not HAS_OCI_PY_SDK:
+        module.fail_json(msg="oci python sdk required for this module.")
```

Comment:
In Ansible 2.8 we added a helpter method method for this https://github.com/ansible/ansible/pull/51787
```suggestion
        module.fail_json(msg=missing_required_lib("oci"))
```

Reply:
done